include/boost/geometry/util/type_traits_std.hpp

```diff
@@ -195,22 +198,21 @@ using transcribe_cref_t = typename transcribe_cref<From, To>::type;
     \endcode
 */
 template <bool IsConst, typename Type>
-struct add_const_if_c
-{
-    typedef std::conditional_t
+struct [[deprecated]] add_const_if_c
+    : std::conditional
         <
             IsConst,
             Type const,
             Type
-        > type;
-};
+        >
+{};
 
 
 namespace util
 {
 
 template <typename T>
-using bare_type = remove_cptrref<T>;
+using bare_type [[deprecated]] = remove_cptrref<T>;
```

Comment:
I think this needs a comment saying why this is deprecated, and what is the replacement.

Reply:
Actually I was not aware of `cptrref` and find it hardly readable.
`bare_type` is clearer.

But anyway, there is `std::decay` tool, is it equivalent? Can't we deprecate both and use that instead (if it works)?

Reply:
I've used this name because in the standard there are:
```
remove_const<>
remove_volatile<>
remove_reference<>
remove_pointer<>
```
and then there are:
```
remove_cv<>
remove_cvref<>
```
The latter first removes reference and then const volatile. Because originally `bare_type` was first removing reference, then pointer, then cv (because there can be a reference to pointer) I named it:
```
remove_cptrref<>
```
If we wanted to remove volatile too which we probably should do I'd call it:
```
remove_cvptrref
```
but originally volatile was not removed so I didn't do it.

AFAIU `std::decay` wouldn't remove pointer.